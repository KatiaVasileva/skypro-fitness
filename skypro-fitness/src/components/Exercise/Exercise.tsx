import { useWorkoutContext } from "../../hooks/useWorkoutContext";
import { ExerciseType } from "../../types/WorkoutType.type";

function Exercise({
  exercise,
//   order,
}: {
  exercise: ExerciseType;
  order: number;
}) {
  const { exercises } = useWorkoutContext();

  console.log(exercises);

//   const progressInPercent = exercises[order]
//     ? (exercises[order].progressWorkout * 100) / exercise.quantity
//     : 0;

  return (
    <div className="pb-5">
      <p className="text-[16px] md:text-[18px]">{exercise.name + " (" + exercise.quantity + " повторений) " + Math.floor(50) + " %"} </p>
      <progress
        className="h-1.5 w-[280px] md:w-[320px] [&::-webkit-progress-bar]:rounded-lg [&::-webkit-progress-bar]:bg-[#F7F7F7] [&::-webkit-progress-value]:rounded-lg [&::-webkit-progress-value]:bg-[#00C1FF]"
        max="100"
        // value={progressInPercent}
        value="50"
      />
    </div>
  );
}

export default Exercise;
