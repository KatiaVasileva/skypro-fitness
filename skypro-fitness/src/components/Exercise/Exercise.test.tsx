/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import * as router from "react-router";
import UserProvider from "../../context/UserContext";
import CoursesProvider from "../../context/CoursesContext";
import { BrowserRouter } from "react-router-dom";
import WorkoutProvider from "../../context/WorkoutContext";
import Exercise from "./Exercise";

const navigate = jest.fn();

describe("Exercise component", () => {
  beforeEach(() => {
    jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
  });

  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <CoursesProvider>
            <WorkoutProvider>
              <Exercise exercise={{
                            name: "",
                            quantity: 0,
                            progressWorkout: 0
                        }} courseId={undefined} workoutId={undefined} />
            </WorkoutProvider>
          </CoursesProvider>
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
