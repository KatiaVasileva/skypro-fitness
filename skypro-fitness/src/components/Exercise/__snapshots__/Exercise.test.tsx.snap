// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise component renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <p
      class="text-[16px] md:text-[18px]"
    >
       (0 повторений) NaN %
       
    </p>
    <progress
      class="h-1.5 w-[280px] md:w-[320px] [&::-webkit-progress-bar]:rounded-lg [&::-webkit-progress-bar]:bg-[#F7F7F7] [&::-webkit-progress-value]:rounded-lg [&::-webkit-progress-value]:bg-[#00C1FF]"
      max="100"
      value="NaN"
    />
  </div>
</div>
`;
